src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/format/RecordWriter.java

```diff
@@ -0,0 +1,32 @@
+package com.microsoft.azure.kusto.kafka.connect.sink.format;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+public interface RecordWriter extends Closeable {
+  /**
+   * Write a record to storage.
+   *
+   * @param record the record to persist.
+   */
+  void write(SinkRecord record) throws IOException;
+
+  /**
+   * Close this writer.
+   */
+  void close();
+
+  /**
+   * Flush writer's data and commit the records in Kafka. Optionally, this operation might also
+   * close the writer.
+   */
+  void commit();
+
+  /**
+   *
+   * @return Total length of file
+   */
+  long getDataSize();
```

Comment:
we dont need this (please look at my comment for zippedBytes  in filewriter.java)

Reply:
We are using getDataSize to get the current size of the file which is required to compare file size with the file threshold  [(code link)](https://github.com/hashedin/kafka-sink-azure-kusto/blob/avro-json-support/src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java#L261) and start flushing the file accordingly. But if we use numBytes from the counting stream than we get the actual file size only after gzipping the file  [(code link)](https://github.com/hashedin/kafka-sink-azure-kusto/blob/avro-json-support/src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java#L136)

Reply:
counting stream counts whatever stream you wrap it around...
so as i suggested in the other comment we can just wrap the gzipStream with the counting stream instead ..

Reply:
We are already wrapping gzip stream with the counting stream [code link](https://github.com/hashedin/kafka-sink-azure-kusto/blob/avro-json-support/src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java#L122). But in gzip stream numbytes will be updated only when  [finish()](https://github.com/hashedin/kafka-sink-azure-kusto/blob/avro-json-support/src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java#L136) method is called which internally writes data to countingOutputStream using write() method where numbytes will be calculated. To overcome this problem, we are manually calculating the file size to compare with [threshold](https://github.com/hashedin/kafka-sink-azure-kusto/blob/avro-json-support/src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java#L261) to check when we should finish the file.

Reply:
and i suggest it will be the other way around -  i think that when a user sets this flush size he will most likely expects that the flush size is over the number of bytes consumed by the connector - not on the intermediate file size .... so we want to count the number of raw bytes passed to the writer .. not the gzipped bytes .. 
also this code - `value.toString().getBytes().length` is too much overhead for no reason

Reply:
what do u say